app/screens/create_or_edit_channel/channel_info_form.tsx

```diff
@@ -161,21 +156,25 @@ export default function ChannelInfoForm({
         setHeaderPosition(nativeEvent.layout.y);
     }, []);
 
+    const onHeaderInputLayout = useCallback(({nativeEvent}: LayoutChangeEvent) => {
+        setHeaderHeight(nativeEvent.layout.height);
+    }, []);
+
     const scrollHeaderToTop = useCallback(() => {
         if (scrollViewRef?.current) {
-            scrollViewRef.current?.scrollToPosition(0, headerPosition);
+            setTimeout(() => scrollViewRef.current?.scrollToPosition(0, headerPosition), 100);
         }
     }, []);
 
     const onKeyboardDidShow = useCallback((frames: any) => {
         setKeyBoardVisible(true);
-        if (Platform.OS === 'android') {
-            setKeyboardHeight(frames.endCoordinates.height);
-        }
-    }, [scrollHeaderToTop]);
+        setKeyboardPosition(frames.endCoordinates.screenY);
+        setKeyboardHeight(frames.endCoordinates.height);
+    }, []);
 
-    const onKeyboardDidHide = useCallback(() => {
+    const onKeyboardDidHide = useCallback((frames: any) => {
```

Comment:
Why are we using any for the frames type?

Reply:
KeyboardAwareScrollView expects an Object, so I did not want to make too many assumptions. But yes, I can put the `KeyboardEvent` type.